### PR TITLE
[quant][graphmode][fx][fix] Speedup all_node_args_have_no_tensors

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#55352 [quant][graphmode][fx][fix] Speedup all_node_args_have_no_tensors**

Summary:
Adding a cache for this function to speedup the computation

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27584279](https://our.internmc.facebook.com/intern/diff/D27584279)

### CODE REVIEW

File: torch/quantization/fx/utils.py [hunk 1]
```diff
@@ -346,31 +346,34 @@ def create_qparam_nodes(quantizer: QuantizerCls, node_name: str, scale: Any, zer
     return (scale_node, zero_point_node)
 
 
-def all_node_args_have_no_tensors(node: Node, modules: Dict[str, torch.nn.Module]) -> bool:
+def all_node_args_have_no_tensors(node: Node, modules: Dict[str, torch.nn.Module], cache: Optional[Dict[Node, bool]] = None) -> bool:
```
Comment (vkuzo) [inline torch/quantization/fx/utils.py:349]: optional: we could also do something like

```
def all_node_args_have_no_tensors(node, modules, cache):
  if node in cache:
    return cache[node]
  result = _all_node_args_have_no_tensors_impl(node, modules)
  cache[node] = result
  return result
```

either is good

**Review by vkuzo — APPROVED** [review]
thanks for speeding this up!  Just curious, what was the magnitude of the speedup?

File: torch/quantization/fx/utils.py (hunk 1 above)
Comment (jerryzh168) [reply torch/quantization/fx/utils.py:349]: thanks, this looks cleaner and less error prone

Comment (jerryzh168) [reply torch/quantization/fx/utils.py:349]: actually, I think we still need to modify the original function to include cache..

**Comment by jerryzh168** [comment]
landed in another PR

**Pull request closed without merging**